AlphaWallet/Core/Formatters/CurrencyFormatter.swift

```diff
@@ -2,47 +2,62 @@
 
 import Foundation
 
-class CurrencyFormatter {
-    static var formatter: NumberFormatter {
-        let formatter = NumberFormatter()
-        formatter.minimumFractionDigits = 2
-        formatter.maximumFractionDigits = 2
-        formatter.roundingMode = .down
-        //TODO support multiple currency values
-        formatter.currencyCode = Currency.USD.rawValue
-        formatter.numberStyle = .currency
+extension NumberFormatter {
 
-        return formatter
+    static let currency = Formatter(.currency)
+    static let usd = Formatter(.usd)
+    static let short = Formatter(.short)
+
+    class Formatter {
+        private let formatter: NumberFormatter
+
+        fileprivate init(_ configuration: NumberFormatterConfiguration) {
+            formatter = configuration.formatter
+        }
+
+        func string(from number: NSNumber) -> String? {
+            return formatter.string(from: number)
+        }
+
+        func string(for: Any?) -> String? {
+            return formatter.string(for: number)
+        }
     }
+}
+
+private enum NumberFormatterConfiguration {
+    case short
+    case usd
+    case currency
 
-    static var usdFormatter: NumberFormatter {
+    var formatter: NumberFormatter {
         let formatter = NumberFormatter()
-        formatter.minimumFractionDigits = 2
-        formatter.maximumFractionDigits = 2
         formatter.roundingMode = .down
-        //TODO support multiple currency values
         formatter.numberStyle = .currency
-        formatter.positiveFormat = "0.00" + " " + Constants.Currency.usd
-        formatter.negativeFormat = "-0.00" + " " + Constants.Currency.usd
 
-        return formatter
-    }
+        switch self {
+        case .currency:
+            formatter.minimumFractionDigits = Constants.formatterFractionDigits
+            formatter.maximumFractionDigits = Constants.formatterFractionDigits
 
-    static var plainFormatter: EtherNumberFormatter {
-        let formatter = EtherNumberFormatter.full
-        formatter.groupingSeparator = ""
-        return formatter
-    }
+            //TODO support multiple currency values
+            formatter.currencyCode = Currency.USD.rawValue
+        case .short:
+            formatter.minimumFractionDigits = Constants.currencyFormatterFractionDigits
```

Comment:
I think if the number of decimals is set to 4 in `CurrencyFormatter`, it has to be moved to `EtherNumberFormatter` and the call site will have to use `EtherNumberFormatter` instead. Example in `SendViewModel`:

```
let value = NumberFormatter.short.string(for: balanceValue)!
```

Similarly EtherNumberFormatter should only have decimals=4 (if there's code that set it to 2, it should be moved to `CurrencyFormatter`.

Reply:
@hboon seems like done, check it.